share/picongpu/tests/CurrentDeposition/lib/python/test/CurrentDeposition/MainTest.py

```diff
@@ -0,0 +1,205 @@
+
+        # write current density tuple
+        J = [J_x, J_y, J_z]
+        Js.append(J)
+        poss.append(pos2)
+        pos_offs.append(pos_off)
+
+    return Js, poss, pos_offs, charge
+
+
+def compare(j_grid_x, j_grid_y, j_grid_z, J):
+    """Compares the current density J provided by the simulation with the
+    values j_grid_x/y/z from the python reference implementation.
+    herefore abs(j(PIConGPU) - j(Python)) is calculated and compared against
+    epsilon = 1e-5, which is slightly higher than PIConGPU numerical
+    uncertainty.
+
+    Parameters:
+    j_grid_x/y/z(np.array): current density calculated by the reference
+    J(np.array): current density calculated by PIConGPU
+
+    Returns:
+    0: if both current densities coincide
+    42: if they don't coincide
+    43: if the masked arrays _not_zero don't have the same shape
+
+    """
+
+    # compare the arrays values by subtracting
+    x_compare = abs(J[0] - j_grid_x)
+    y_compare = abs(J[1] - j_grid_y)
+    z_compare = abs(J[2] - j_grid_z)
+
+    # error boundary; slightly higher than numerical uncertainty
+    epsilon = 1e-5
+
+    # are the reference and simulated values the same? (within error margin)
+    if np.all(x_compare < epsilon) and np.all(y_compare < epsilon) and np.all(z_compare < epsilon):
+        print("simulation and reference coincide")
+        return 0
+    else:
+        print("no consensus between simulation and reference")
+        print("please check the simulation")
+        return 42
+
+
+def main(dataPath):
+    """Main function.
+
+    Parameters:
+    dataPath(string): path to the simulation data
+
+    Returns:
+    compare_result(int): 0 if the simulated result can be regarded equal to the
+                         reference
+                         !0 if they differ.
+
+    """
+    # read parameters of series
+    series, order, params = get_params(dataPath)
+
+    # read current densities, charge, positions
+    Js, poss, pos_offs, charge = read_series(series)
+    # read the size of the current density array
+    shape = np.shape(Js[1][0])
+
+    # create grid_object for comparison
+    compare_grid = grid(order)
+
+    grid_x, grid_y, grid_z = compare_grid.create_grid()
+    start_koord, end_koord = compare_grid.particle_step(poss[0], poss[1], pos_offs[0], pos_offs[1])
+    # computation of current density by python reference implementation
+    W_grid_x, W_grid_y, W_grid_z = compare_grid.current_deposition_field(start_koord, end_koord, grid_x, grid_y, grid_z)
+
+    j_grid_x, j_grid_y, j_grid_z = compare_grid.current_density_field(
+        W_grid_x, W_grid_y, W_grid_z, start_koord, end_koord, charge[0], params, shape
+    )
+
+    # comparison
+    compare_result = compare(j_grid_x, j_grid_y, j_grid_z, Js[1])
+
+    return compare_result
+
+
+if __name__ == "__main__":
+    if len(sys.argv) != 2:
+        raise SystemExit(f"Usage: {sys.argv[0]} <path_to_simulation_data>")
+    main(sys.argv[1])
```

Comment:
CAUTION: Otherwise the test never fails:
```suggestion
    sys.exit(main(sys.argv[1]))
```

Reply:
PS: Have you ever tried to make the test fail? Do you actually know that it does the correct thing?

Reply:
I triggered the test to fail as I changed it to the current status of how to compare the two results. I added a `+1` to the current density. The correct thing happened and I got the Error message.

Reply:
Yes, there's no doubt that you would get a message printed to screen. Question is: What was the exit code of of `bin/ci.sh`? Once we start having runtime tests in CI again, that is the only automatically checked indicator for failure and that exit code is -- for all I understand -- currently 0.

Reply:
The CurrentDeposition Test now outputs the right code in the case of failure (42) and success (0).